_blog/2024-02-09-using-clerk-in-rust.mdx

```diff
@@ -0,0 +1,1014 @@
+---
+title: "Using Clerk authentication in Rust"
+description: This article is on how to use Clerk authentication in Rust based back-end applications and little bit of an extension on integration of shared database using Shuttle. 
+author: sourabpramanik
+tags: [rust, clerk, vite, guide]
+thumb: 
+cover: 
+date: 
+---
+
+In this article, I have shared how I built an Issue tracking application using Vite for the front-end, and [Shuttle](https://www.shuttle.rs/) and [Actix web](https://actix.rs/) for the back-end. In addition to that I have also used [Clerk](https://clerk.com/) for authentication in the front-end and protecting the endpoints in the back-end.
+
+Here is the [source code](https://github.com/sourabpramanik/issue-tracker) of the complete project if you want to follow along.
+
+To start this project I have used the template from Shuttle’s [example repository](https://github.com/shuttle-hq/shuttle-examples) that has a barebone setup of Vite, Shuttle, Actix web, and Clerk. 
+
+The back-end uses [clerk-rs](https://github.com/DarrenBaldwin07/clerk-rs) crate to create a connection between the back-end application and Clerk project. This crate has many different APIs for different use cases, in the template there is back-end API to fetch all users record using `clerk-rs` user API. We can easily verify user's identity trying to access protected routes before hand using the `ClerkMiddleware` from the crate.
+
+I have additionally used Postgres for the database which will run in a docker container so make sure you also have docker [installed](https://docs.docker.com/engine/install/).
+
+
+# Let's get started
+
+First, we need to clone the template from the Shuttle’s example repository using Shuttle CLI:
+
+```bash
+cargo shuttle init --from shuttle-hq/shuttle-examples --subfolder actix-web/clerk
+
+```
+
+You will get a few prompts to create a new project using this template. Make sure you provide a unique name because when you deploy this project on Shuttle this project name will become the sub-domain. And once a name is taken it cannot be used for other project as well:
+
+```bash
+What do you want to name your project?
+It will be hosted at ${project_name}.shuttleapp.rs, so choose something unique!
+✔ Project name · <ENTER_PROJECT_NAME>
+Where should we create this project?
+✔ Directory · <ENTER_THE_DIRECTORY_NAME>
+✔ Claim the project name <PROJECT_NAME> by starting a project container on Shuttle? <Y/N>
+
+```
+
+After that, `cd` into the project directory:
+
+```bash
+cd {project_name}
+
+```
```

Comment:
This can also be left out. The sentence above is verbose enough.

Reply:
Cool!